searchcore/src/vespa/searchcore/proton/server/bucketmovejobv2.cpp

```diff
@@ -234,34 +242,59 @@ BucketMoveJobV2::prepareMove(BucketMoverSP mover, std::vector<MoveKey> keys, IDe
 void
 BucketMoveJobV2::completeMove(BucketMoverSP mover, std::vector<GuardedMoveOp> ops, IDestructorCallbackSP onDone) {
     mover->moveDocuments(std::move(ops), std::move(onDone));
+    handleMoveResult(std::move(mover));
+}
+
+void
+BucketMoveJobV2::handleMoveResult(BucketMoverSP mover) {
     if (mover->bucketDone() && mover->inSync()) {
-        _modifiedHandler.notifyBucketModified(mover->getBucket());
+        BucketId bucket = mover->getBucket();
+        assert(_bucketsInFlight.contains(bucket));
+        _modifiedHandler.notifyBucketModified(bucket);
+        _bucketsInFlight.erase(bucket);
+        updatePending();
+        if (_postponedUntilSafe.contains(bucket)) {
+            reconsiderBucket(_ready.meta_store()->getBucketDB().takeGuard(), bucket);
```

Comment:
Missing a `_postponedUntilSafe.erase(bucket)` before reconsiderBucket ?

Reply:
Good point eternal work is never good.